Community Toolbox/scripts/utils_CommunityToolboxDsList/utils_CommunityToolboxDsList.gml

```diff
@@ -0,0 +1,47 @@
+/// @func ds_list_create_from_array(array)
+/// @desc Creates a new DS list with items from the given array.
+/// @arg {Array} array      The array to get the items from.
+/// @returns {Id.DsList}
+function ds_list_create_from_array(_array) {
+    var _list = ds_list_create();
+    ds_list_add_from_array(_list, _array);
+    return _list;
+}
+
+/// @func ds_list_add_from_array(list,array)
+/// @desc Adds items from the given array to the given DS list.
+/// @arg {Id.DsList} list       The list to add the items to.
+/// @arg {Array} array          The array to get the items from.
+function ds_list_add_from_array(_list, _array) {
+    var _length = array_length(_array);
```

Comment:
While it doesn't look like there's any `ds_list_resize` function similar to `array_resize`, it does look like you can use `ds_list_set` to force the capacity of the list to be correct in one go. Assuming `ds_list` is backed by a dynamically-sized array buffer (similar to C++'s `std::vector`), then pre-allocating with `ds_list_set` will be more performant than adding each element by hand, since the latter approach can cause multiple resizes.

In the case where the array is non-empty, consider a `ds_list_set(_list, ds_list_size(_list) + _length - 1, undefined /* Placeholder value */)`, or equivalent code with `[| ...]` accessors, before the loop starts.